lib/schnorr.py

```
@@ -0,0 +1,101 @@
+        except:
+            continue
+        if secp256k1:
+            break
+    if not secp256k1:
+        print_stderr('[schnorr] warning: libsecp256k1 library failed to load')
+        return None
+
+    try:
+        secp256k1.secp256k1_context_create.argtypes = [c_uint]
+        secp256k1.secp256k1_context_create.restype = c_void_p 
+
+        secp256k1.secp256k1_context_randomize.argtypes = [c_void_p, c_char_p]
+        secp256k1.secp256k1_context_randomize.restype = c_int
+
+        secp256k1.secp256k1_schnorr_sign.argtypes = [ c_void_p, c_char_p, c_char_p, c_char_p, c_char_p, c_char_p]
+        secp256k1.secp256k1_schnorr_sign.restype = c_int 
+
+        secp256k1.ctx = secp256k1.secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY)
+        r = secp256k1.secp256k1_context_randomize(secp256k1.ctx, os.urandom(32))
+        if r:
+            return secp256k1
+        else:
+            print_stderr('[schnorr] warning: secp256k1_context_randomize failed')
+            return None
+    except (OSError, AttributeError):
+        #traceback.print_exc(file=sys.stderr)
+        print_stderr('[schnorr] warning: libsecp256k1 library was found and loaded but there was an error when using it')
+        return None 
+
+try:
+    _libsecp256k1 = load_library()
+except:
+    _libsecp256k1 = None
+    #traceback.print_exc(file=sys.stderr)
+
+def sign_schnorr(message,privkey):
+    
+    # Error handling in case of libsec failure:
+    wefailed= False 
+    try:
+        _libsecp256k1 = load_library()
+    except:
+        _libsecp256k1 = None
+        wefailed= True
+    if wefailed:
+        return bytes.fromhex("00")  #Will be handled by reversion to ECDSA upon return
+   
+    # Normally, proceed...
+    secp256k1=load_library()
+    schnorr_signature = create_string_buffer(64)
+    dummyvariable = _libsecp256k1.secp256k1_schnorr_sign(secp256k1.ctx,schnorr_signature,message,privkey,None,None)
```

it's not a dummy variable -- need to throw an exception on failure!